packages/superset-ui-encodable/src/utils/combineContinuousDomains.ts

```diff
@@ -0,0 +1,39 @@
+import { isEveryElementDefined, isDefined } from '../typeGuards/Base';
+
+/**
+ * Combine two continuous domain and ensure that the output
+ * does not go beyond fixedDomain
+ * @param fixedDomain
+ * @param inputDomain
+ */
+export default function combineContinuousDomains(
+  fixedDomain: (number | Date | null | undefined)[],
+  inputDomain?: (number | Date)[],
+) {
+  if (fixedDomain.length > 0 && isEveryElementDefined(fixedDomain)) {
+    return fixedDomain;
+  } else if (inputDomain) {
+    if (
+      fixedDomain.length === 2 &&
+      inputDomain.length === 2 &&
+      fixedDomain.filter(isDefined).length > 0
+    ) {
+      const [min1, max1] = fixedDomain;
```

Comment:
maybe rename `min1, max1` to something along the lines of  `userSpecifiedMin` and `userSpecifiedMax` and `min2, max2` to `datasetMin`, and `datasetMax` for better readability